lib/walk.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+module.exports.walk = function walk(rootNode, visitor, exitVisitor) {
```

Comment:
This was two months ago now, but I vaguely remember encountering some issue with `estree-walk`. But I can't remember what the reason was, and agree it'd be better to use something off the shelf for this.

Reply:
I think it's because `estree-walker` V3 dropped support for CJS. We can use V2 for now and migrate to V3 once we drop support for CJS. In the next major version for example.

Reply:
After spending not insignificant amount of time, I realized that `estree-walker` does not skip the `parent` property, but visits it and then visits its children which again contain `parent` back to it. This results in an infinite loop.